src/types.ts

```diff
@@ -1376,6 +1377,33 @@ export type FeatureFlagsCallback = (
     }
 ) => void
 
+export type FeatureFlagDetail = {
+    key: string
+    enabled: boolean
+    // Only used when overriding a flag payload.
+    original_enabled?: boolean | undefined
+    variant: string | undefined
+    // Only used when overriding a flag payload.
+    original_variant?: string | undefined
+    reason: EvaluationReason | undefined
+    metadata: FeatureFlagMetadata | undefined
+}
+
+export type FeatureFlagMetadata = {
+    id: number
+    version: number | undefined
+    description: string | undefined
+    payload: JsonType | undefined
+    // Only used when overriding a flag payload.
+    original_payload?: JsonType | undefined
+}
+
+export type EvaluationReason = {
+    code: string
```

Comment:
the code isn't super human-readable (e.g. `condition_match` or `no_condition_match`, seems like a low lift to have a helper that renders better strings in the UI. This is probably a posthog app change (taxonomy), but in e.g. the persons view we have this mapping to show human-friendly match reasons
```
const featureFlagMatchMapping = {
    [FeatureFlagMatchReason.ConditionMatch]: 'Matches',
    [FeatureFlagMatchReason.NoConditionMatch]: "Doesn't match any conditions",
    [FeatureFlagMatchReason.OutOfRolloutBound]: 'Out of rollout bound',
    [FeatureFlagMatchReason.NoGroupType]: 'Missing group type',
    [FeatureFlagMatchReason.SuperConditionMatch]: 'Matches early access condition',
    [FeatureFlagMatchReason.Disabled]: 'Disabled',
}

```

Reply:
Agreed. I updated it to just send the reason.description.